### DISCUSSION TITLE
How to enable the `update.TIMESTAMP` measure?

### DISCUSSION THREAD

**Discussion by sjmg** [atoti/atoti#635] [Q&A] [answered]
Hello,

We work in activePivot frequently and we have in this technology the meassure timestamp. We don't know if this meassure is created automatically in Atoti. Exist this meassure? or exist an example creating this meassure?

Thanks so much

**Comment by CamilleBlanco** [comment]
Hi 😄 👋 !

It is indeed automatically created in atoti too, but not visible by default. You can make it visible with `cube.measures["update.TIMESTAMP"].visible=True` 👀

**Reply by mwebborn-av** [reply]
Hi,
Should we expect this measure to be blank if we haven't configured an aggregate provider? Not sure if Atoti uses JIT or Leaf by default?

**Comment by sjmg** [comment]
Hello,

We do it the changes that you told us. But this meassures don't show us a result. It's necessary to add this folder or any configuration for read this meassure?

Thanks

**Comment by bloussou** [comment, marked as answer]
Hello,
To have values in `update.TIMESTAMP` you need to set an aggregate provider. Does the previous project have such providers ? In the next release of atoti a [ublic way to set such providers will be provided like (example from the tutorial of the documentation): 
```python
cube.aggregate_providers.update(
        {
            "test provider": tt.AggregateProvider(
                key="bitmap",
                levels=[l["Date"]],
                measures=[m["Quantity.SUM"]],
            )
        }
```
And then I'm able to see values for the measure:
![image](https://user-images.githubusercontent.com/32448554/180779972-26c48224-e48d-4a8c-8f6e-3062b261dc6b.png)

    )

**Reply by HuifangYeo** [reply]
Hi,

why do we need to set aggregate provider to see the timestamp? Do we always set it as bitmap?

**Reply by bloussou** [reply]
It is like this in ActivePivot. [Copper.timestamp()](https://github.com/activeviam/activepivot/blob/6.0/pivot/copper2/impl/src/main/java/com/activeviam/copper/api/Copper.java#L84) is returning a [TimestampFunction](https://github.com/activeviam/activepivot/blob/6.0/composer/aggregation/src/main/java/com/qfs/agg/impl/TimestampFunction.java#L25) and the doc says `This aggregation function gives you the last time the aggregator has been updated.`. So if no aggregator is provided (the default in atoti) then this measure is null
